cosmos/operators/watcher.py

```diff
@@ -282,6 +315,13 @@ def poke(self, context: Context) -> bool:
             self.model_unique_id,
         )
 
+        producer_task_state = ti.xcom_pull(task_ids=self.producer_task_id, key="state")
+
+        if producer_task_state == "failed":
+            raise AirflowException(
+                f"""The dbt build command failed in producer task. Please check the log of task {self.producer_task_id} for details."""
+            )
```

Comment:
It may happen that the producer tasks reported failure because only few of the models/resources failed running but the rest of the models succeeded. So, my strong suggestion is that we check for this xcom only once the corresponsing model's status is checked first. And if that xcom is not available then fail the sensor.

Reply:
I have moved it down to check after model's status is checked